src/sidebar/components/tag-editor.js

```diff
@@ -1,35 +1,197 @@
 'use strict';
 
-// @ngInject
-function TagEditorController(tags) {
-  this.onTagsChanged = function() {
-    tags.store(this.tagList);
-
-    const newTags = this.tagList.map(function(item) {
-      return item.text;
-    });
-    this.onEditTags({ tags: newTags });
+const { createElement } = require('preact');
+const propTypes = require('prop-types');
+const { useMemo, useRef, useState } = require('preact/hooks');
+
+const { withServices } = require('../util/service-context');
+const SvgIcon = require('./svg-icon');
+
+/**
+ * Component to edit annotation's tags.
+ */
+function TagEditor({ annotation, onEditTags, tags: tagsService, tagList }) {
+  const inputEl = useRef(null);
+  const [showSuggestions, setShowSuggestions] = useState(false);
+
+  // List of suggestions returned from the tagsService
+  const suggestions = useMemo(() => {
+    // Remove any repeated suggestions that are already tags.
+    const removeDuplicates = (suggestions, tags) => {
+      const suggestionsSet = [];
+      suggestions.forEach(suggestion => {
+        if (tags.indexOf(suggestion) < 0) {
+          suggestionsSet.push(suggestion);
+        }
+      });
+      return suggestionsSet.sort();
+    };
+    // Call filter with an empty string to return all suggestions
+    return removeDuplicates(tagsService.filter(''), tagList);
+  }, [tagsService, tagList]);
+
+  /**
+   * Handle changes to this annotation's tags
+   */
+  const updateTags = tagList => {
+    // update suggested tags list via service
+    tagsService.store(tagList.map(tag => ({ text: tag })));
+    // invoke callback with updated tags
```

Comment:
I think the next line of code is obvious enough you could leave this comment out.